app/controllers/BankDetailsController.scala

```diff
@@ -98,7 +98,20 @@ class BankDetailsController @Inject() (
   }
 
   val checkBankDetails: Action[AnyContent] = as.eligibleJourneyAction { implicit request =>
-    Ok("This is where the check bank details page will go, for now, here's the journey:\n\n\n" + Json.prettyPrint(Json.toJson(request.journey)))
+    request.journey match {
+      case j: Journey.BeforeEnteredDirectDebitDetails => JourneyIncorrectStateRouter.logErrorAndRouteToDefaultPage(j)
+      case j: Journey.AfterEnteredDirectDebitDetails =>
+        if (j.directDebitDetails.isAccountHolder) {
+          Ok(views.bankDetailsSummary(j.directDebitDetails, BankDetailsController.enterBankDetailsUrl))
+        } else {
+          Redirect(routes.BankDetailsController.cannotSetupDirectDebitOnlinePage().url)
+        }
+    }
+  }
+
+  val checkBankDetailsSubmit: Action[AnyContent] = as.eligibleJourneyAction.async { implicit request =>
```

Comment:
we should probably put that check on the stage on the submit endpoint as well - it'd be bad if we allowed someone to use this POST endpoint to bypass that check and proceed anyway. Not sure if anyone actually would but better to be safe

Reply:
will raise a ticket, I assume you mean in the backend?

Reply:
we'll probably be able to bring it into this sprint tbh

Reply:
no I meant around the bit of code before we talk to the backend here

Reply:
Done